### DISCUSSION TITLE
Question about 'on pull_request' build workflow behavior

### DISCUSSION THREAD

**Discussion by omindu** [community/community#25823] [Actions]
<p>I have a workflow which triggers a build when a PR is opened and the workflow is similar to below.</p>
<pre><code>on:
   pull_request:
      branches:
         - master
...
jobs:
  build:
    steps:
    - uses: actions/checkout@v1
      with:
        ref: master
</code></pre>
<p>My question is, when we create a pull request from a forked repository to the main repository, is the triggered build will be done on the main repository branch (master branch in this case) after applying the pull request ?</p>
<p>In my case, it doesn’t seem like the build happens with the PR content applied. The reason is, I created a PR with a fix for a unit test failure, but the triggered build’s test case failed due to not having that fix. However once the PR was mergerd, the subsequent PR builds were successful.</p>
<p>Highly appriciate some input on the build behavior.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>If you add <strong>with.ref</strong> , it will check out the specific branch. In your example, it checked out master branch. It didn’t include your pull request content.</p>
<p>Please remove <strong>with.ref</strong> from actions/checkout@v1</p>
<pre><code>with:
   ref: master
</code></pre>
<p>      </p>
<p>Just use</p>
<pre><code>- uses: actions/checkout@v1
</code></pre>
<p>By default, the branch or tag ref that triggered the workflow will be checked out. If the workflow is triggered by pull request, then <strong>refs/remotes/pull/[id]/merge</strong> will be checkout.</p>
<p>  </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7bd5abc7b5bed597513e6feaad17a608428f1cc4.png" data-download-href="/uploads/short-url/hFuw2Z99NjJ3ko6nTBD7nuQxGCM.png?dl=1" title="Pull request checkout.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181093632-cfdc0ddc-1d79-4e15-8feb-ed6b3a79dda9.png" alt="Pull request checkout.png" data-base62-sha1="hFuw2Z99NjJ3ko6nTBD7nuQxGCM" width="567" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7bd5abc7b5bed597513e6feaad17a608428f1cc4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Pull request checkout.png</span><span class="informations">733×646 32.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>For more detail info, please refer to Checkout Action document &gt; <a href="https://github.com/actions/checkout#usage" rel="nofollow noopener">https://github.com/actions/checkout#usage</a></p>

**Comment by omindu** [comment]
<p>Thanks for the detailed answer. I have done the suggested changes.</p>
